### ISSUE TITLE
download rules

### ISSUE THREAD

**Issue by richfitz** [richfitz/remake#34]
Downloading files to start an analysis is likely to be fairly common.

@sckott found a cool makefile rule that uses curl to check if a URL has likely changed content: https://ryanfb.github.io/etc/2015/02/25/url_dependencies_with_make.html

Implementing this in R should be easier than the convolutions in make, especially as we have the RCurl package and things like httr.  Ideally it would be possible to write

```
targets:
  filename:
    download: http://whatever.com/path
```

and apply the recipe above.  In addition, we should probably not freak out if the network is not available (I *am* currently in Australia), and just give a warning that the check was not done.

**Comment by sckott** [comment]
not sure how to contribute. you using R6 classes?

**Comment by richfitz** [comment]
Yes, but this function should not need to care about this.  If we can replicate the download_url function I can organise saving whatever bits of metadata into remake's various stores and passing them into the function.

Looks to me like we need a url, destination filename, last-modified (from the server) and "etags".  So all I need is a function that takes those as arguments, will always download if the last two are `NULL` but otherwise does the header magic.  I guess `httr` or `RCurl` could do this?

**Comment by sckott** [comment]
which pkg would you rather have as a dep? `RCurl` or `httr`

**Comment by richfitz** [comment]
`httr` I think, even though it depends on `RCurl` - I think that the `RCurl` bit is going to get subbed out for the new libcurl wrapper soon.

**Comment by sckott** [comment]
here's what I got so far  https://gist.github.com/sckott/0a18d89c0df97dee8d55
